### PR TITLE
feat: merged cell related documents, tests and some refactoring

### PR DESCRIPTION
**Pull request by stone-lyl**
### 👀 PR includes

✨ Feature

- [X] New feature

🎨 Enhance

- [ ] Code style optimization
- [X] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [X] Test case
- [X] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description
1. 添加获取默认选择单元格作为合并单元格信息的功能。
2. 将合并单元格和取消合并单元格通过 interaction 文件对外抛出。
3. 添加合并单元格文档。
4. 更新合并单元格demo
5. 添加合并单元格功能的部分单测。
### 🖼️ Screenshot

| Before                         | After                         |
| ------------------------------ | ------------------------------ |
| ❌                             | ✅                              |

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/common/interface/interaction.ts [hunk 1]
```diff
@@ -13,6 +20,7 @@ export type S2CellType<T extends SimpleBBox = ViewMeta> =
   | ColCell
   | CornerCell
   | RowCell
+  | MergedCells
```
Comment (lijinke666) [inline packages/s2-core/src/common/interface/interaction.ts:23]: MergedCell ?

File: packages/s2-core/src/utils/interaction/merge-cells.ts [hunk 2]
```diff
@@ -190,17 +211,33 @@ const getCellsByInfo = (
...
   sheet: SpreadSheet,
-  cellsInfo: MergedCellInfo[],
+  cellsInfo?: MergedCellInfo[],
   hideData?: boolean,
 ) => {
+  let mergeCellsInfo = cellsInfo;
+  if (!mergeCellsInfo) {
+    mergeCellsInfo = getActiveCellsInfo(sheet);
+  }
+
```
Comment (lijinke666) [inline packages/s2-core/src/utils/interaction/merge-cells.ts:221]:
```suggestion
const mergeCellsInfo = cellsInfo || getActiveCellsInfo(sheet)

```

**Review by lijinke666 — APPROVED** [review]

File: packages/s2-core/src/common/interface/interaction.ts (hunk 1 above)
Comment (stone-lyl) [reply packages/s2-core/src/common/interface/interaction.ts:23]: ok

File: packages/s2-core/src/utils/interaction/merge-cells.ts [hunk 3]
```diff
@@ -244,18 +278,21 @@ const removeUnmergedCellsInfo = (
...
-    if (newMergedCellsInfo?.length !== sheet.options?.mergedCellsInfo?.length) {
-      sheet.setOptions({
-        mergedCellsInfo: newMergedCellsInfo,
-      });
-      removedCells.remove(true);
-    }
+export const unmergeCell = (sheet: SpreadSheet, removedCells: MergedCell) => {
+  if (!removedCells || removedCells.cellType !== CellTypes.MERGED_CELLS) {
+    // eslint-disable-next-line no-console
+    console.error('unmergeCell: the cell is not a MergedCell');
```
Comment (wjgogogo) [inline packages/s2-core/src/utils/interaction/merge-cells.ts:284]: 可以把 `removedCells`打印一下
```suggestion
    console.error(`unmergeCell: the cell ${removedCells.xxx} is not a MergedCell`);
```

**Review by xingwanying — DISMISSED** [review]
test修下，以及动态渲染的判断条件有些小问题
BUG

File: packages/s2-core/src/utils/interaction/merge-cells.ts (hunk 3 above)
Comment (xingwanying) [reply packages/s2-core/src/utils/interaction/merge-cells.ts:284]: log信息都走 `DebuggerUtil`  类

**Comment by lijinke666** [comment]
需要再加两个 RootInteraction 的测试 

https://github.com/antvis/S2/blob/535a7929cc53a2f892e476ab87b418e6fe31adcd/packages/s2-core/__tests__/unit/interaction/root-spec.ts#L87-L92

**Comment by stone-lyl** [comment]
> test修下，以及动态渲染的判断条件有些小问题 BUG

已修复

**Comment by stone-lyl** [comment]
明细表meta

**Comment by stone-lyl** [comment]
> 需要再加两个 RootInteraction 的测试
> 
> https://github.com/antvis/S2/blob/535a7929cc53a2f892e476ab87b418e6fe31adcd/packages/s2-core/__tests__/unit/interaction/root-spec.ts#L87-L92

已经加了

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**